Sources/TuistLoader/Mappers/ProjectDescriptionHelpersLocationMapper.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  ProjectDescriptionHelpersLocationMapper.swift
+//  TuistLoader
+//
+//  Created by Luis on 11/10/20.
+//
+
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistSupport
+
+/// Mapper that changes the development region based on configuration
+/// in the Config.swift
+public final class ProjectDescriptionHelpersLocationMapper: ProjectMapping {
+    private let config: TuistCore.Config
+    private let rootDirectoryLocator: RootDirectoryLocating
+
+    private lazy var derivedHelpersDirectory: AbsolutePath? = {
+        guard let configPath = config.path,
+              let rootDirectory = rootDirectoryLocator.locate(from: configPath) else {
+            return nil
+        }
+
+        return rootDirectory
+            .appending(component: Constants.tuistDirectoryName)
+            .appending(component: Constants.helpersDirectoryName)
+            .appending(component: Constants.DerivedDirectory.name)
+    }()
+
+    public init(
+        config: TuistCore.Config,
+        rootDirectoryLocator: RootDirectoryLocating = RootDirectoryLocator()
+    ) {
+        self.config = config
+        self.rootDirectoryLocator = rootDirectoryLocator
+    }
+
+    // MARK: - ProjectMapping
+
+    public func map(project: TuistCore.Project) throws -> (TuistCore.Project, [SideEffectDescriptor]) {
+        guard let helpersLocation = config.helpersLocation else {
+            return (project, [])
+        }
+
+        logger.notice("Using \(helpersLocation) for ProjectDescriptionHelpers")
+
+        switch helpersLocation {
+        case let .local(path):
+            return (project, try collectLocalHelpers(at: path))
+        case let .gitWithBranch(url, gitId),
+             let .gitWithHash(url, gitId),
+             let .gitWithTag(url, gitId):
+            return (project, try collectGitHelpers(at: url, gitId: gitId))
+        }
+    }
+
+    /// Collects files from a local path and copies them to the `ProjectDescriptionHelpers` location.
+    private func collectLocalHelpers(at path: String) throws -> [SideEffectDescriptor] {
+        guard let derivedHelpersDirectory = derivedHelpersDirectory else {
+            logger.error("Unable to locate derived helpers directory.")
+            return []
+        }
+
+        let localHelpersPath = AbsolutePath(path)
+
+        let sources = try FileHandler.shared
+            .contentsOfDirectory(localHelpersPath)
+            .filter { $0.extension == "swift" }
+
+        let sourceContents = try sources.map { try FileHandler.shared.readFile($0) }
+        let sourceNames = sources.map(\.basename)
+        let createHelpersDirectory = DirectoryDescriptor(path: derivedHelpersDirectory, state: .present)
+
+        let createFiles = zip(sourceNames, sourceContents)
```

Comment:
`zip` works, but maybe we can create a new `FileCopyDescriptor` that will do the copying instead? Copying is a common operation, so it'd be great if we can reuse that logic in the future.

Reply:
Agree, added!